app/src/models/stash-entry.ts

```diff
@@ -7,4 +9,16 @@ export interface IStashEntry {
 
   /** The SHA of the commit object created as a result of stashing. */
   readonly stashSha: string
+
+  /** The list of files this stash touches */
+  readonly files: StashedFileChanges
 }
+
+export enum StashedChangesLoadStates {
+  NotLoaded,
+  Loading,
+}
+
+export type StashedFileChanges =
```

Comment:
This type feels like it's trying to accomplish two distinct things, how do you feel about making this an interface?

Ex.
```typescript
inteface ICouldNotThinkOfAGoodName {
  readonly files: ReadonlyArray<CommittedFileChanges>
  readonly loadState: StashedChangesLoadState
}
```

Reply:
Go with https://github.com/desktop/desktop/pull/7213#discussion_r274547593